### ISSUE TITLE
[AI] Surrogate v3.2.0 trained on synthetic data with `expected_accuracy = 0.0` — no held-out EnergyPlus hourly validation, surrogate drift gate cannot detect real-world failure

### ISSUE THREAD

**Issue by anchapin** [anchapin/fluxion#2458]
**Severity:** High
**Affected Cases:** All ASHRAE 140 cases in the production surrogate path (`ThermalModelMode::Hybrid` with `use_surrogate_loads = true`, the default per `HybridRouting::default()`). On the 2026-08-08 working tree, the v3.2.0 model is the only ONNX artifact in the registry (`tests/surrogate_models/registry.json`), and every `predict_loads_with_fallback` call on a loaded-v3.2.0 SurrogateManager returns its predictions.
**Blocked By:** None — but unblocks the surrogate drift tolerance gate CI (#1865, #1784, #2208 chain) which currently can pass or fail on synthetic-vs-physics equivalence without the surrogate ever being exercised against real measurement data.
**Architecture Contract:** `SurrogateManager::predict_loads_with_fallback` (`src/ai/surrogate.rs:1784-1817`); `SurrogateInputs::from_physics` / `from_physics_data` (`src/ai/surrogate.rs:268-316`); the surrogate training pipeline contract in `data/training/README.md` (10,000 samples minimum for production surrogate v3.1, 50,000 for hardened CI gate); the model registry contract in `tests/surrogate_models/registry.json` (each model entry pins `training_data_hash` + `training_data_provenance`).

### Gap Description

`tests/surrogate_models/registry.json` (the canonical model registry, regenerated
by `scripts/gen_registry.py`) records for v3.2.0:

```json
{
  "version": "3.2.0",
  "model_sha256": "eafdb8d567d11e99acd883ab437721c360fd7db53d931c8f99527e2afb76fd15",
  "training_data_hash": "bae4d13711464778888edd132fa1efd920c438622489f9962c6c3d11ad8969bc",
  "trained_on": "2026-07-14",
  "training_data_provenance": {
    ...
    "training_data_hash_source": "SHA-256 of data/synthetic/v2.1 directory (exact source not independently verified in this audit)",
    "audit_note": "expected_accuracy is 0.0 because no held-out test set was used during training. TODO: run validate_surrogate.py to establish a real accuracy baseline."
  }
}
```

Two failures pinned by the registry itself:

1. **`expected_accuracy: 0.0`** — the model was not evaluated against a held-out test set. The `validate_surrogate.py` script (`scripts/validate_surrogate.py:1-379`) exists but is not run as part of the model acceptance criteria. The model is shipped as "production" with no measured accuracy.
2. **`training_data_hash_source: "data/synthetic/v2.1"`** — the training data is **synthetic**, not extracted from EnergyPlus simulations. The `data/training/README.md` makes the contract explicit: "Physics is the only legitimate training source" and the cloud-campaign pipeline (`scripts/cloud_campaign_manager.py`) is required to produce real `(X, y)` pairs. The `data/training/` directory contains only the README — no `samples_*.csv` files exist on disk.

The `surrogate_zone_thermal_metrics.json` (the v3.0 model metadata, still
committed at `models/`) records:

```json
{
  "model_version": "3.0.0",
  "training_samples": 800,
  "test_samples": 200,
  "test_rmse": 0.5487794413304624,
  "test_mae": 0.43250107765197754,
  "test_r2": 0.9955677390098572,
  "rmse_normalized": 0.015489791377497534,
}
```

The 800-sample training set is **13× below** the 10,000-sample production
threshold in `data/training/README.md`. The R² of 0.996 is on the 200-sample
in-sample test split — not held-out. The `train_test_split` was a random
shuffle of the synthetic dataset, so the test set is leakage-adjacent.

### Impact

- The surrogate drift tolerance gate (`surrogate_drift_gate.yml` per `.github/workflows/surrogate_drift_gate.yml`) compares the surrogate output to the physics output. If both the physics engine and the surrogate are wrong in the same direction (e.g., both undercount south-window solar in September), the drift gate stays green while the hybrid output is wrong. ARCHITECTURE.md Phase 3 (§ML Surrogate Drop-In) explicitly states "Surrogates must match physics within 2% on held-out data" — this gate is not enforced.
- Users running large parametric sweeps via `BatchOracle::evaluate_population` (the 16.80 cases/sec production path per `docs/ASHRAE140_RESULTS.md`) get surrogate predictions to within 2% of an *unverified* target. The first user who queries against ASHRAE 140 Case 940 (night setback, described in Issue #1 of `tmp/issues_physics.md`) will see the 5–10× annual energy over-prediction propagate through the surrogate.
- The `data/training/` directory is empty. Even though the training pipeline is wired correctly (per `tools/train_surrogate.py::load_training_data` which refuses to run without `samples_*.csv`), the infrastructure to *produce* the samples (`scripts/cloud_campaign_manager.py`) requires cloud credentials that are not available to CI. The pipeline is correctly refusing to fall back to synthetic per Issue #1338, which means the pipeline is currently blocked.

### First Step

Verify the gap is real (it is, per the audit_note in the registry), then
close the loop in two layers that do NOT require cloud credentials:

1. **Measure the in-sample-only quality honestly**: add a `--force-honest-eval` flag to `scripts/validate_surrogate.py` that:
   - Loads the on-disk v3.2.0 ONNX model.
   - Splits the synthetic training data by `case_id` (not random) — so the test set holds out entire ASHRAE 140 cases (e.g., 940) that the surrogate has never seen.
   - Reports per-case MAE and percent deviation.
   - Asserts the publish criterion: `max per-case MAE < 2%` (per ARCHITECTURE.md Phase 3 rule). If the assertion fails, the registry entry is annotated with `production_blocked: true` and the model is replaced by `predict_loads_with_fallback`'s sine cycle (`SurrogateManager::deterministic_analytical_loads`) at runtime.
2. **Wire the honest-eval into CI**: add a job to `.github/workflows/surrogate_drift_gate.yml` that runs `python scripts/validate_surrogate.py --force-honest-eval --model models/surrogate_v3.2.0.onnx` and fails the workflow if the per-case MAE exceeds 2%. The current drift gate (vs physics, not vs held-out) stays as a *secondary* check.

Per RULES.md, do NOT produce a synthetic "training set" to make the eval pass.
The fix is to make the eval honest — the production model may be demoted to
"benchmark only" status, which is the correct outcome given the registry's own
audit_note.

### Notes

- Related issues: #1865 (Surrogate Drift Tolerance Gate fails — no trained ONNX model loaded in CI, closed but the underlying gap is real), #1784 (drift gate >1%, closed), #1338 (closed — the synthetic fallback was removed from `tools/train_surrogate.py`), #1139 (v3.0 baseline training).
- The `SurrogateManager::deterministic_analytical_loads` sine-cycle fallback (`src/ai/surrogate.rs:1845-1853`) is a real, deterministic safety net. If the honest-eval demotes v3.2.0, the engine still runs — just slower and with sinusoidal-saturated loads. This is preferable to a silently-wrong ML model.
- ARCHITECTURE.md Phase 3 says "Surrogates must match physics within 2% on held-out data" — the current release gates do not enforce this. Adding it as a release gate is the goal.
- Out of scope: retraining against a real EnergyPlus sweep (requires cloud credentials). The honest-eval step is the prerequisite gate that must be green before any future re-train can be considered production-ready.
